remote/blueprints/image.py

```diff
@@ -163,18 +209,53 @@ def resize_avatar():
     # Resize it to each size contained in `AvatarSize`.
 
     start = time.time()
+    avatars = {}
     try:
         # Confirmation basic avatar to resize.
         # Try to use max size avatar otherwise use original(rotated).
-        base_avatar = handler.auto_rotated().image
-        if _standard_avatar := _try_rescale(base_avatar, max(AvatarSize)):
-            base_avatar = ImageHandle.load_from_bytes(_standard_avatar)
-
-        avatars = {
-            f"avatar{size}": avatar_summary
-            for size in AvatarSize
-            if (avatar_summary := _get_avatar_summary(base_avatar, size))
-        }
+        if not handler.animated:
+            handler.auto_rotate()
+
+        image = handler.image
+        for size in AvatarSize:
+            if size.is_mandatory or _need_rescale(image, size):
+                # Choose rescale function based on format and animation.
+                if handler.animated and request.args.get("animated"):
+                    if handler.mime == ImageMIME.GIF or handler.mime == ImageMIME.WEBP:
+                        frames, durations = _rescale_animated_image_frames(
+                            image, size, resizeimage.resize_cover, validate=False
+                        )
+                    elif handler.mime == ImageMIME.PNG:
+                        frames, durations = _rescale_animated_png_frames(
+                            image, size, resizeimage.resize_cover, validate=False
+                        )
+                    else:
+                        frames = None
+
+                    if frames:
+                        data = _save_animated_frames_data(
+                            frames,
+                            durations,
+                            format=ImageMIME.PNG.pil_format,
+                            loop=0,  # always loop for avatar
```

Comment:
Not sure whether it is better to keep the loop attr of the original or always loop, what do you think?

Reply:
It is better to keep animated avatars looping.